src/analyze.jl

```diff
@@ -157,10 +159,91 @@ function _slic(alg::SLIC, img::AbstractArray{<:Lab, 2})
         segments ./= n_segment_elems # broadcast: (n_segments,) -> (n_segments, 5)
     end
 
+    # Enforce connectivity
+    # Reference: https://github.com/scikit-image/scikit-image/blob/7e4840bd9439d1dfb6beaf549998452c99f97fdd/skimage/segmentation/_slic.pyx#L240-L348
     if alg.enforce_connectivity
-        nothing # TODO
+        println("zergling")
+        segment_size = height * width / n_segments
+        min_size = round(Int, 0.5 * segment_size)
+        max_size = round(Int, 3.0 * segment_size)
+
+        dy = [1, -1, 0, 0]
+        dx = [0, 0, 1, -1]
+        # dz = [] # reversed for supervoxels
+
+        start_label = 1
+        mask_label = start_label - 1 # indicates the label of this pixel has not been assigned
+        nearest_segments_final = fill(mask_label, height, width)
+        current_new_label = start_label
+
+        # used for BFS
+        current_segment_size = 1
+        bfs_visited = 0
+
+        # store neighboring pixels
+        # now set the dimension to 2 because we are using superpixel
+        coord_list = fill(0, max_size, 2)
+ 
+        for x = 1:width
+            for y = 1:height
+                if nearest_segments[y, x] == mask_label continue end
+                if nearest_segments_final[y, x] > mask_label continue end
```

Comment:
In Julia we usually use

```suggestion
                nearest_segments[y, x] == mask_label && continue
                nearest_segments_final[y, x] > mask_label && continue
```